src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -2348,7 +2348,7 @@ lambdaParameter is not SourceComplexParameterSymbolBase
                     }
 
                     // If synthesizing a delegate with an [UnscopedRef] parameter, check the attribute is available.
-                    if (delegateParameter.HasUnscopedRefAttribute)
+                    if (delegateParameter.HasUnscopedRefAttribute && delegateParameter.UseUpdatedEscapeRules)
```

Comment:
>delegateParameter.UseUpdatedEscapeRules [](http://example.com/codeflow?start=69&length=39)

It looks like we're still emitting the `[UnscopedRef]` attribute in `SynthesizedParameterSymbol.AddSynthesizedAttributes()` regardlelss of `UseUpdatedEscapeRules`. Is that expected?

Reply:
Are any tests affected by this change? If not, consider adding a test similar to `SynthesizedDelegateTypes_UnscopedRefAttribute_MissingConstructor_01` which seems to test the call below when `UseUpdatedEscapeRules` is set.

Reply:
> Are any tests affected by this change?

I think this one cannot be tested - `HasUnscopedRefAttribute` will only be `true` if `UseUpdatedEscapeRules` is `true` since the lambda and the delegate live in the same compilation. That being said, I'm reverting the change here for the same reason as in https://github.com/dotnet/roslyn/pull/75930#discussion_r1856799309.